components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -134,15 +141,17 @@ public String[] registerOAuthConsumer() throws IdentityOAuthAdminException {
      */
     public OAuthConsumerAppDTO[] getAllOAuthApplicationData() throws IdentityOAuthAdminException {
 
-        String userName = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        String userName;
         OAuthConsumerAppDTO[] dtos = new OAuthConsumerAppDTO[0];
-
-        if (userName == null) {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            userName = OAuthUtil.getUsername(tenantDomain);
```

Comment:
can this `username` be null? If so, do a null check using stringUtils

Reply:
Null user scenario is handled using and exception. Therefore a null check is not required here.